src/driver/mpas_subdriver.F

```diff
@@ -109,38 +111,72 @@ end subroutine xml_stream_get_attributes
       readNamelistArg = .false.
       readStreamsArg = .false.
 
-      nArgs = command_argument_count()
-      iArg = 1
-      do while (iArg < nArgs)
-         call get_command_argument(iArg, argument)
-         if (len_trim(argument) == 0) exit
-
-         if ( trim(argument) == '-n' ) then
-            iArg = iArg + 1
+      ! If provided, error check the namelistFileParam and copy it to namelistFile to override default
+      if (present(namelistFileParam)) then
+         if (len_trim(namelistFileParam) <= 0) then
+            write(0,"(A)") 'WARNING: mpas_init argument namelistFileParam has 0 or negative length and will be ignored'
+         else if (len_trim(namelistFileParam) > len(namelistFile)) then
+            write(0,"(A)") 'CRITICAL ERROR: mpas_init argument namelistFileParam is too long for namelistFile variable.'
+            write(0,"(A,I5,A,I5)") 'len_trim(namelistFileParam)=', len_trim(namelistFileParam), &
```

Comment:
It's a minor issue, but we've generally tended to use single-quotes whenever possible (one exception is when writing strings that contain single quotes, where it can sometimes be clearer to enclose the entire string in double-quotes).

Reply:
That's a standard I can stick to. Thanks!